File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        
        ListNode slow = head;
        ListNode fast = head.next;
        
        ArrayList<Integer> stack = new ArrayList<Integer>(0);
        
        while (fast != null){
            stack.add(slow.val);
            
            slow = slow.next;
            fast = fast.next;

            if (fast != null) fast = fast.next;
        }
        
        int maxSum = 0;

        while (slow != null){
            maxSum = Math.max(maxSum, slow.val + stack.remove(stack.size()-1));
            slow = slow.next;
        }
        

        return maxSum;
        
    }
}

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        
        slowPointer, fastPointer = head, head.next
        
        stack = []
        
        while fastPointer:
            stack.append(slowPointer.val)
            
            slowPointer = slowPointer.next
            fastPointer = fastPointer.next
            if fastPointer:
                fastPointer = fastPointer.next

        maxSum = 0
        while slowPointer:
            maxSum = max(maxSum, stack.pop() + slowPointer.val)
            slowPointer = slowPointer.next
        
        
        return maxSum
            
        